File: attendancemanagement.client/src/consts/types/timeZoneConst.ts
/** types */
import ConstBase from "./constBase"

/** タイムゾーン情報定義 */
export default class TimeZoneConst extends ConstBase {

    /** ロケール */
    private _local: string;

    /** タイムゾーン */
    private _timeZone: string;

    /**
     * コンストラクタ
     * @param id ID
     * @param locale ロケール
     * @param timeZone タイムゾーン
     */
    public constructor(id: string, local: string, timeZone: string) {
        super(id);
        this._local = local;
        this._timeZone = timeZone;
    }

    /// <summary>
    /// ロケール
    /// </summary>
    public get locale(): string {
        return this._local;
    }

    /// <summary>
    /// タイムゾーン
    /// </summary>
    public get timeZone(): string {
        return this._timeZone;
    }
}

File: WebApiService/Common/class/TimeZoneConst.cs
using Common.Const;

namespace WebApiService.Common.Const
{
    /// <summary>
    /// タイムゾーン情報
    /// </summary>
    public class TimeZoneConst : ConstBaseWithDictionary<TimeZoneConst>
    {
        /// <summary>
        /// コンストラクタ
        /// </summary>
        /// <param name="id">ID</param>
        /// <param name="timeZone">タイムゾーン</param>
        internal TimeZoneConst(string id, string timeZone) 
            : base(id)
        {
            this.TimeZone = timeZone;
            this.TimeZoneInfo = TimeZoneInfo.FindSystemTimeZoneById(timeZone);
            Dictionary.Add(id, this);
        }

        /// <summary>
        /// タイムゾーン
        /// </summary>
        public string TimeZone { get; private set; }

        /// <summary>
        /// タイムゾーン情報
        /// </summary>
        public TimeZoneInfo TimeZoneInfo { get; private set; }
    }
}